numpy/lib/arraypad.py

```diff
@@ -888,7 +888,17 @@ def pad(array, pad_width, mode='constant', **kwargs):
         length = _as_pairs(length, padded.ndim, as_index=True)
         for axis, width_pair, length_pair in zip(axes, pad_width, length):
             roi = _view_roi(padded, original_area_slice, axis)
-            stat_pair = _get_stats(roi, axis, width_pair, length_pair, func)
+            try:
+                stat_pair = _get_stats(
+                    roi, axis, width_pair, length_pair, func)
+            except ValueError as error:
```

Comment:
Would it makes since to push this within `_get_stats`, and only catch the `ValueError`s from the calls to `stat_func`?

Could do something like the following, as an inner closure of `_get_stats`:
```
def better_error_stat_func(chunk, axis, keepdims):
    try:
        return stat_func(chunk, axis, keepdims)
    except ValueError as error:
        # as below
```

Reply:
Good suggestion. However `get_stats` potentially calls `stat_func` twice. To avoid duplication I wrapped the content of the whole function and do an explicit check if the error is the one we wan't to "replace".